src/tiledimage.js

```diff
@@ -1227,6 +1244,18 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
         }
     },
 
+    /**
+     * Enable cache preservation even without this tile image,
+     * by default disabled. It means that upon removing,
+     * the tile cache does not get immediately erased but
+     * stays in the memory to be potentially re-used by other
+     * TiledImages.
+     * @param {boolean} allow
+     */
+    allowZombieCache: function(allow) {
```

Comment:
You are missing cases where people force system to throw away data while planinng to add the same tiled image back again right away. You might remember `replace` prop in the `addTiledImage` method, or see here https://github.com/usnistgov/OpenSeadragonFiltering/blob/f1e350c08e8bf050fdc84b89e9b04ad041c861b9/openseadragon-filtering.js#L211.  Basically, cache keeps only data that has at least one living tile. But even data with non-living tile might be (and many times are) useful. People had to re-download all the data due to lack of support for async processing, like we have now. Althoug it is no longer necessary to throw away all downloaded data to do some async updates, it still might bring performance improvements.

Reply:
But it made me think what happens when the zombie gets loaded back again. I guess we don't execute the invalidation pipeline on it, though we probably should, as invalidation routine skips zombies.

Reply:
> Basically, cache keeps only data that has at least one living tile.

I'd argue that a cache system should keep anything with a living tile plus however many it is configured to keep in the actual *cache* (i.e. the thing that keeps old data around for a while) and that this should be the default behavior. If someone wants to force it to throw away data, there could be an API for that. Having the default be throwing away data when a `TiledImage` is unloaded doesn't feel right to me. That would be like a browser clearing the cache for a webpage as soon as the page is navigated to a different domain.

Also, in case a new `TiledImage` is loaded with the same source url, the cache key should be based on the tile url plus whatever info indicates the data processing state (rather than an instance of a `Tile` object). That way, multiple `TiledImage`s can share cache data (even if they aren't open at the same time).

Reply:
> I'd argue that a cache system should keep anything with a living tile plus however many it is configured to keep in the actual cache

Yes, this is the new feature called 'zombies'.

> Having the default be throwing away data when a TiledImage is unloaded doesn't feel right to me.

sticking to the original behavior before this feature was available. Can be changed. RN it is set automatically and temporarily ON only when system recognizes you call `addTiledImage` with swap, and `prev.tileSource.equals(next.tileSource)` holds.

> Also, in case a new TiledImage is loaded with the same source url, the cache key should be based on the tile url plus whatever info indicates the data processing state (rather than an instance of a Tile object). That way, multiple TiledImages can share cache data (even if they aren't open at the same time).

I don't understand. Tile cache keys are arbitrary, and if they equal, the cache is reused among tiles regardless of what source they come from. We call it `cache-equal` tiles and `getTileCache` governs the behavior.

Reply:
Just a few more notes:

>  That would be like a browser clearing the cache for a webpage as soon as the page is navigated to a different domain.

Browsers do this. `sessionStorage`. It is also configurable whether you want to have this on or off. Maybe you are confused by the separation of the internal cache objects, but implementation-wise, this is easier approach since you keep two arrays of caches - one has references, one does not. It fits to the old behavior, and you can opt out from this new feature. Ond from the touside, the cache behaves exactly as you described, it is just split internally, + it offers API to control support for the feature.

> in case a new TiledImage is loaded with the same source url, the cache key should be based on the tile url plus whatever info indicates the data processing state (rather than an instance of a Tile object)

If this is the case, the tile is not downloaded at all, it is just attached to the existing cache and inherits the state of all tiles that share that cache. This way we avoid any needless processing. True, if someone created custom cache atop what the system internally does, it might collide, the system might try to inherit state of a tile based just on some third-party tile reference. But we repeat this in the docs all over the place. If people play with tile cache, they must ensure unique key per unique data.